.github/workflows/release.yml

```diff
@@ -64,165 +82,61 @@ jobs:
         env:
           JF_ENV_1: ${{ secrets.ZITI_ARTIFACTORY_CLI_CONFIG_PACKAGE_UPLOAD }}
 
-      - name: Upload RPM to Artifactory release repo
+      - name: Upload RPM to Artifactory pre-release repo
         shell: bash
-        run: |
-          for RPM in ./dist/*.rpm; do
-            echo "INFO: Uploading $RPM"
-            jf rt upload --recursive=false --flat=true \
-              "$RPM" \
-              ${{ env.ZITI_RPM_PROD_REPO }}/redhat/x86_64/
-          done
-
-      - name: Upload DEB to Artifactory release repo
-        shell: bash
-        run: |
-          for DEB in ./dist/*.deb; do
-            echo "INFO: Uploading $DEB"
-            jf rt upload --recursive=false --flat=true \
-            --deb=debian/main/amd64 \
-            "$DEB" \
-            ${{ env.ZITI_DEB_PROD_REPO }}/pool/zrok/amd64/
-          done
-
-  build-linux-arm64:
-    runs-on: ubuntu-20.04
-    steps:
-      - run: sudo apt update
-      - run: sudo apt-get install gcc-aarch64-linux-gnu
-
-      - uses: actions/checkout@v4
-        with:
-          fetch-depth: 0
-
-      - run: git fetch --force --tags
-
-      - uses: actions/setup-go@v5
-        with:
-          go-version-file: ./go.mod
-          cache: true
-
-      - uses: actions/setup-node@v4
-        with:
-          node-version: 18.x
-
-      - run: npm install
-        working-directory: ui
-
-      - run: npm run build
-        working-directory: ui
-        env:
-          CI: "true"
-
-      - uses: goreleaser/goreleaser-action@v6
-        with:
-          distribution: goreleaser
-          version: '~> v2'
-          args: release --skip=publish --config .goreleaser-linux-arm64.yml
-        env:
-          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-
-      - uses: actions/upload-artifact@v4
-        with:
-          name: release-builds-linux-arm64
-          path: ./dist/*.gz
-
-      - name: Configure jFrog CLI
-        uses: jfrog/setup-jfrog-cli@v4
-        with:
-          version: ${{ env.JFROG_CLI_VERSION }}
         env:
-          JF_ENV_1: ${{ secrets.ZITI_ARTIFACTORY_CLI_CONFIG_PACKAGE_UPLOAD }}
-
-      - name: Upload RPM to Artifactory release repo
-        shell: bash
+          ARCH: ${{ matrix.arch.rpm }}
         run: |
-          for RPM in ./dist/*.rpm; do
-            echo "INFO: Uploading $RPM"
-            jf rt upload --recursive=false --flat=true \
-              "$RPM" \
-              ${{ env.ZITI_RPM_PROD_REPO }}/redhat/aarch64/
+          set -o xtrace
+          shopt -s nullglob
```

Comment:
I refactored the shell scripts that upload Linux packages to Artifactory to be more strict and fail fast if a package is unexpectedly missing, etc.